AndroidAnnotations/androidannotations-core/androidannotations/src/main/java/org/androidannotations/internal/core/handler/RootFragmentHandler.java

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright (C) 2016-2019 the AndroidAnnotations project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed To in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.androidannotations.internal.core.handler;
+
+import static com.helger.jcodemodel.JExpr.cast;
+import static com.helger.jcodemodel.JExpr.lit;
+import static org.androidannotations.helper.LogHelper.logTagForClassHolder;
+
+import javax.lang.model.element.Element;
+import javax.lang.model.type.TypeMirror;
+
+import org.androidannotations.AndroidAnnotationsEnvironment;
+import org.androidannotations.ElementValidation;
+import org.androidannotations.annotations.RootFragment;
+import org.androidannotations.handler.BaseAnnotationHandler;
+import org.androidannotations.handler.MethodInjectionHandler;
+import org.androidannotations.helper.InjectHelper;
+import org.androidannotations.holder.EBeanHolder;
+
+import com.helger.jcodemodel.AbstractJClass;
+import com.helger.jcodemodel.IJAssignmentTarget;
+import com.helger.jcodemodel.IJExpression;
+import com.helger.jcodemodel.JBlock;
+import com.helger.jcodemodel.JConditional;
+import com.helger.jcodemodel.JInvocation;
+
+public class RootFragmentHandler extends BaseAnnotationHandler<EBeanHolder> implements MethodInjectionHandler<EBeanHolder> {
+
+	private final InjectHelper<EBeanHolder> injectHelper;
+
+	public RootFragmentHandler(AndroidAnnotationsEnvironment environment) {
+		super(RootFragment.class, environment);
+		injectHelper = new InjectHelper<>(validatorHelper, this);
+	}
+
+	@Override
+	public void validate(Element element, ElementValidation validation) {
```

Comment:
I was wondering we should allow only Fragment extending types, however i see that you deliberately want to inject custom interfaces as well. `@RootContext` only allows types extending `Context` to be injected. I think we should do similar here.

Reply:
Well, we could definitely check if it extends Fragment (support one, or system one)... I guess the annotation `@RootFragment` would be more clear then... I definitely would like to inject interfaces, though I could live with not being able to do it on this annotation ;)

In fact, I added an annotation `@RootView`  to the Plugin I'm building, which basically will use the "fragment or context activity" past to the EBean, and that one will support using Interfaces then. 

From an architecture point of view, it makes sense then, it would be quite usefull for implementing MVP and MVVM (though with some improvements to avoid holding a real instance of the activity/fragment on the beans). 

Well, this `@RootView` could be implemented directly on AA, but I think it will make more sense on this Plugin handling architectural patterns, but let me know if you think is useful on AA itself.

Reply:
Let's require `Fragment` here then.